src/server/kernel/wxwaylandsurface.cpp

```diff
@@ -457,22 +459,34 @@ WXWaylandSurface::DecorationsType WXWaylandSurface::decorationsType() const
     return static_cast<DecorationsType>(d->nativeHandle()->decorations);
 }
 
+void WXWaylandSurface::setSurfaceSizeRatio(qreal ssr)
+{
+    Q_D(WXWaylandSurface);
+
+    if (qFuzzyCompare(ssr, d->surfaceSizeRatio))
+        return;
+    d->surfaceSizeRatio = ssr;
+}
+
 bool WXWaylandSurface::checkNewSize(const QSize &size)
 {
+    Q_D(WXWaylandSurface);
+
     const QSize minSize = this->minSize();
     const QSize maxSize = this->maxSize();
 
+    QSize overrideScaleSize = size * d->surfaceSizeRatio;
```

Comment:
No, should check the WSurfaceItem::resizeSurface return value in tinywl-qtquick.
Because the `surfaceSizeRatio` is working with WSurfaceItem, isn't be relevant to WXWaylandSurface.